src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from "../../../../common/format";
+
+class Tier20_4p extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  totalAimed = 0;
+  buffedCastAndFocusAimed = 0;
+  //buffedCastAimed = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_4P_BONUS.id);
+  }
+  get percentUptime() {
+    //This calculates the uptime over the course of the encounter of the 4 p bonus
+    const uptime = this.combatants.selected.getBuffUptime(SPELLS.HUNTER_MM_T20_4P_BONUS_BUFF.id) / this.owner.fightDuration;
+
+    return uptime;
+  }
+
+  /*
+  // Irrelevant for now, hoping to find a way to register when a cast is only affected by tier at start of cast, because the way it functions is two-part.
+  on_byPlayer_begincast(event) {
+    if(this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_4P_BONUS_BUFF.id) && (event.ability.guid === SPELLS.AIMED_SHOT.id)) {
+      this.buffedCastAimed += 1;
+    }
+  }*/
```

Comment:
You should probably remove commented code like this (maybe make an issue on your fork repository as a reminder of this code so it doesn't get entirely lost?). Same with the commented line 15.